### PR TITLE
Complete Review Functionality

### LINKED ISSUE #43: As a user, I want to review all the cards in a deck
- [x] Create a Review class
- [x] Create ReviewCommand
- [x] Create EndReviewCommand
- [x] Create FlipCommand
- [x] Create NextCommand
- [x] Create PreviousCommand
- [x] Create CorrectCommand
- [x] Create WrongCommand
- [x] Implement Logic for the above commands

### PR DESCRIPTION
**Pull request by unfazing**
Added IsSameCardPredicate - used to filter the Card currently under review so that it can be displayed in GUI.
Implemented the FlipCardCommand, NextCardCommand, PreviousCardCommand, MarkCorrectCommand, MarkWrongCommand and their respective logic. 
The review is reflected in the GUI :-)

Notes:
- In the GUI, question number in the reviewed card needs to be removed (currently it is always 1.)
- The commands for review functions look weird but are all the nearest keys to the enter key for user convenience
  - flip `[`
  - previous `]`
  - next `\`
  - correct `;`
  - wrong `'`
- Currently the user can start reviewing a deck at anytime. When they end the review, the previous state of the GUI before the review is started will be returned (ie. selectedDeck and filteredCardList unchanged after review).
- The user cannot review an empty deck.
- The user can go to the next or previous card or end the review regardless if the current card has been flipped.
- The user must flip a card before they can mark it as correct or wrong.
- To implement in future - a custom message for next / previous card when there are no more cards.

closes #43

### CODE REVIEW

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 1]
```diff
@@ -45,7 +45,11 @@ public PersonCard(Card card, int displayedIndex) {
         this.card = card;
         id.setText(displayedIndex + ". ");
         name.setText(card.getQuestion().question);
-        address.setText(card.getAnswer().answer);
+        if (card.isFlipped()) {
+            address.setText(card.getAnswer().answer);
+        } else {
+            address.setText("");
```
Comment (ruiyigan) [inline src/main/java/seedu/address/ui/PersonCard.java:51]: Is this the hacky way O.o

**Review by ruiyigan — APPROVED** [review]
LGTM! Thanks for the work. Left very minor comments. The only other comment I have is the user commands for the different functions. I don't think we should do it based on proximity to the enter button but this one can be discussed another time. Thank youuu

File: src/main/java/seedu/address/ui/PersonCard.java (hunk 1 above)
Comment (unfazing) [reply src/main/java/seedu/address/ui/PersonCard.java:51]: yep.

File: src/main/java/seedu/address/model/ModelManager.java [hunk 2]
```diff
@@ -237,7 +273,13 @@ public Optional<Review> getReview() {
 
     @Override
     public void endReview() {
+        currReview.flipAllCards();
         currReview = null;
+        if (selectedDeck != null) {
```
[the remark this answers is not in the document]
Comment (unfazing) [reply src/main/java/seedu/address/model/ModelManager.java:278]: yep, will need @rockman007372 input on the optionals...

[the remark this answers is not in the document]
Comment (rockman007372) [reply src/main/java/seedu/address/model/ModelManager.java:279]: I think we can just merge now so we can finish the Milestone! Can optimize more in the next milestone!

**Pull request merged**